core/scale/scale_decoder_stream.hpp

```diff
@@ -191,6 +191,8 @@ namespace kagome::scale {
         vec.resize(item_count);
       } catch (const std::bad_alloc &) {
         common::raise(DecodeError::TOO_MANY_ITEMS);
+      } catch (...) {
+        common::raise(DecodeError::UNKNOWN_EXCEPTION);
```

Comment:
I guess is better to re-throw exception to save original exception info.
```suggestion
        throw;
```